wear/src/main/java/com/eveningoutpost/dexdrip/Models/Bubble.java

```diff
@@ -0,0 +1,163 @@
+import com.eveningoutpost.dexdrip.UtilityModels.PersistentStore;
+import com.eveningoutpost.dexdrip.UtilityModels.Pref;
+
+import java.nio.ByteBuffer;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static com.eveningoutpost.dexdrip.xdrip.gs;
+
+/**
+ *
+ */
+
+public class Bubble {
+    private static final String TAG = "Bubble";//?????"Bubble";
+
+    private static volatile byte[] s_full_data = null;
+    private static volatile int s_acumulatedSize = 0;
+
+    public static boolean isBubble() {
+        final ActiveBluetoothDevice activeBluetoothDevice = ActiveBluetoothDevice.first();
+        if (activeBluetoothDevice == null || activeBluetoothDevice.name == null) {
+            return false;
+        }
+        return activeBluetoothDevice.name.contentEquals("Bubble");
+    }
+
+
+    public static BridgeResponse getBubbleResponse() {
+        final BridgeResponse reply = new BridgeResponse();
+        ByteBuffer ackMessage = ByteBuffer.allocate(6);
+        ackMessage.put(0, (byte) 0x02);
+        ackMessage.put(1, (byte) 0x01);
+        ackMessage.put(2, (byte) 0x00);
+        ackMessage.put(3, (byte) 0x00);
+        ackMessage.put(4, (byte) 0x00);
+        ackMessage.put(5, (byte) 0x2B);
+        reply.add(ackMessage);
+        return reply;
+    }
+
+    public static int lens = 344;
+    public static int BUBBLE_FOOTER = 8;
+
+    static int errorCount = 0;
+
+
+
+    public static BridgeResponse decodeBubblePacket(byte[] buffer, int len) {
+        final BridgeResponse reply = new BridgeResponse();
+        int first = 0xff & buffer[0];
+        if (first == 0x80) {
+            PersistentStore.setString("Bubblebattery", Integer.toString(buffer[4]));
+            Pref.setInt("bridge_battery", buffer[4]);
+            PersistentStore.setString("BubbleHArdware", Integer.toString(buffer[2]) + ".0");
+            PersistentStore.setString("BubbleFirmware", Integer.toString(buffer[1]) + ".0");
+            ByteBuffer ackMessage = ByteBuffer.allocate(6);
+            ackMessage.put(0, (byte) 0x02);
+            ackMessage.put(1, (byte) 0x01);
+            ackMessage.put(2, (byte) 0x00);
+            ackMessage.put(3, (byte) 0x00);
+            ackMessage.put(4, (byte) 0x00);
+            ackMessage.put(5, (byte) 0x2B);
+            reply.add(ackMessage);
+            s_full_data = null;
+            Log.e(TAG, "reset data");
+            return getBubbleResponse();
+        }
+        if (first == 0xC0) {
+            String SensorSn = LibreUtils.decodeSerialNumberKey(Arrays.copyOfRange(buffer, 2, 10));
+            PersistentStore.setString("LibreSN", SensorSn);
+            return reply;
+        }
+        if (first == 0x82) {
+            int expectedSize = lens + BUBBLE_FOOTER;
+            if (s_full_data == null) {
+                InitBuffer(expectedSize);
+            }
+            addData(buffer);
+            PersistentStore.setBoolean("CobbleNoSensor", false);
```

Comment:
please define this as a single string at the top of the class and reference it instead of the string literal each time, eg `private static final String COBBLE_NO_SENSOR = "CobbleNoSensor";`

Reply:
Fixed